js/zb.js

```diff
@@ -716,12 +714,122 @@ module.exports = class zb extends Exchange {
+            'swap': 'contractV2PrivateGetFundBalance',
+        });
+        const request = {
+            // 'futuresAccountType': 1, // SWAP
+            // 'currencyId': currency['id'], // SWAP
+            // 'currencyName': 'usdt', // SWAP
+        };
+        if (marketType === 'swap') {
+            request['futuresAccountType'] = 1;
+        }
+        const response = await this[method] (this.extend (request, query));
+        //
+        // Spot
+        //
+        //     {
+        //         "result": {
+        //             "coins": [
+        //                 {
+        //                     "isCanWithdraw": "true",
+        //                     "canLoan": false,
+        //                     "fundstype": 51,
+        //                     "showName": "ZB",
+        //                     "isCanRecharge": "true",
+        //                     "cnName": "ZB",
+        //                     "enName": "ZB",
+        //                     "available": "0",
+        //                     "freez": "0",
+        //                     "unitTag": "ZB",
+        //                     "key": "zb",
+        //                     "unitDecimal": 8
+        //                 },
+        //             ],
+        //             "version": 1645856691340,
+        //             "base": {
+        //                 "auth_google_enabled": true,
+        //                 "auth_mobile_enabled": false,
+        //                 "trade_password_enabled": true,
+        //                 "username": "blank@gmail.com"
+        //             }
+        //         },
+        //         "leverPerm": true,
+        //         "otcPerm": false,
+        //         "assetPerm": true,
+        //         "moneyPerm": true,
+        //         "subUserPerm": true,
+        //         "entrustPerm": true
+        //     }
+        //
+        // Swap
+        //
+        //     {
+        //         "code": 10000,
+        //         "data": [
+        //             {
+        //                 "userId": "6896693805014120448",
+        //                 "currencyId": "6",
+        //                 "currencyName": "usdt",
+        //                 "amount": "30.56585118",
+        //                 "freezeAmount": "0",
+        //                 "contractType": 1,
+        //                 "id": "6899113714763638819",
+        //                 "createTime": "1644876888934",
+        //                 "modifyTime": "1645787446037",
+        //                 "accountBalance": "30.56585118",
+        //                 "allMargin": "0",
+        //                 "allowTransferOutAmount": "30.56585118"
+        //             },
+        //         ],
+        //         "desc": "操作成功"
+        //     }
+        //
         // todo: use this somehow
         // let permissions = response['result']['base'];
-        return this.parseBalance (response);
+        if (marketType === 'swap') {
```

Comment:
We don't really have to do this string comparison twice, let's shrink it down:

```JavaScript
if (marketType === 'swap') {
}
// ...
if (marketType === 'swap') {
}
```
↓
```JavaScript
const swap = (marketType === 'swap');
if (swap) {
}
if (swap) {
}
```